### ISSUE TITLE
Fix request: false positive for NonComparableSetElements - explicit Comparator

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#470]
Not bad when explicit Comparator is provided in constructor:

```java
private static final SortedSet<Class<?>> TYPES = new TreeSet<>(Comparator.comparing(Class::getName))
```
